src/app-layer-ssl.c

```diff
@@ -490,35 +496,65 @@ static int TlsDecodeHSCertificate(SSLState *ssl_state, SSLStateConnp *connp,
             goto next;
         }
 
-        char *str = SCX509GetSubject(x509);
-        if (str == NULL) {
+        SCX509GetSubject(x509, &subject_name, &subject_len);
```

Comment:
Philippe's last comment was not honored. Used `connp->obj` directly instead of another alloc